build_projects/shared-build-targets-utils/Utils/HostVersion.cs

```diff
@@ -106,7 +106,7 @@ public override string ToString()
         // These versions are used when generating platform installers.
         //
         public bool IsLocked = true; // Set this variable to toggle muxer locking.
-        public VerInfo LockedHostFxrVersion => IsLocked ? new VerInfo(1, 0, 1, "", "", "", CommitCountString) : LatestHostFxrVersion;
+        public VerInfo LockedHostFxrVersion => IsLocked ? new VerInfo(1, 0, 2, "", "", "", CommitCountString) : LatestHostFxrVersion;
```

Comment:
Reverted the change to LockedHostFxrVersion to 1.0.1. Now the HostFXR MSI name is back to "dotnet-hostfxr-win-x86.1.0.1.msi" and the folder in which the hostfxr installs is updated to "1.0.5". 

Since, 1.0.5 is a new folder the scenario #4 above is fixed.